zkvm/src/types.rs

```diff
@@ -354,6 +353,15 @@ impl DataWitness {
             DataWitness::Input(b) => b.encode(buf),
         }
     }
+
+    fn serialized_len(&self) -> usize {
+        match self {
+            DataWitness::Program(instr) => instr.iter().map(|p| p.serialized_length()).sum(),
+            DataWitness::Input(b) => 32 + b.contract.serialized_length(),
+            // All other types are encoded as 32-bit points or scalars
+            _ => 32 as usize,
```

Comment:
I’d avoid “_” here and enumerate them all. So that when we add a new type we are not &@$&/:$SEGV.

Reply:
Ideally, i’d call serlen on each type and define them to return 32 within the scope of each type - so things are properly localized.